api/src/main/java/org/pmiops/workbench/firecloud/FireCloudServiceImpl.java

```diff
@@ -315,23 +309,36 @@ public void removeOwnerFromBillingProject(
   @Override
   public FirecloudWorkspace createWorkspace(String projectName, String workspaceName) {
     WorkspacesApi workspacesApi = workspacesApiProvider.get();
-    FirecloudWorkspaceIngest workspaceIngest = new FirecloudWorkspaceIngest();
-    workspaceIngest.setName(workspaceName);
-    workspaceIngest.setNamespace(projectName);
-    checkAndAddRegistered(workspaceIngest);
+    FirecloudWorkspaceIngest workspaceIngest =
+        new FirecloudWorkspaceIngest()
+            .namespace(projectName)
+            .name(workspaceName)
+            .authorizationDomain(
+                ImmutableList.of(
+                    new FirecloudManagedGroupRef()
+                        .membersGroupName(configProvider.get().firecloud.registeredDomainName)));
+
     return retryHandler.run((context) -> workspacesApi.createWorkspace(workspaceIngest));
   }
 
   @Override
   public FirecloudWorkspace cloneWorkspace(
       String fromProject, String fromName, String toProject, String toName) {
     WorkspacesApi workspacesApi = workspacesApiProvider.get();
-    FirecloudWorkspaceIngest workspaceIngest = new FirecloudWorkspaceIngest();
-    workspaceIngest.setNamespace(toProject);
-    workspaceIngest.setName(toName);
-    checkAndAddRegistered(workspaceIngest);
+    FirecloudWorkspaceRequestClone cloneRequest =
+        new FirecloudWorkspaceRequestClone()
+            .namespace(toProject)
+            .name(toName)
+            // We copy only the notebooks/ subdirectory as a heuristic to avoid unintentionally
+            // performing a deep copy of workspace blobs.
```

Comment:
Do we ever want to do a deep copy?

Reply:
I updated the comment, it didn't quite say what I originally intended. Deep copy is ambiguous terminology in this context.

If your question is: are there use cases for copying different subsets of files (including all / none) when copying  a workspace? Yes. Whether we support those is a product question.

Reply:
👍